### PR TITLE
docs: Add some documentation details for VFIO

### LINKED ISSUE #1079: Add details regarding how devices passed with VFIO can be accessed with a container

### PR DESCRIPTION
**Pull request by amshinde**
Add explanation regarding how devices passed with VFIO can
be accessed with a container.

Fixes #1079

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

### CODE REVIEW

**Comment by amshinde** [comment]
@egernst Can you take a look. I am planning to move this Kata eventually.

**Review by mcastelino — COMMENTED** [review]
@amshinde we need to pass -v /dev:/dev so that the VM's /dev is passed into the container /dev right.

Do we filter out any of the /dev entries? Or for that matter just bind mount the /dev corresponding to the device to the container?

**Comment by amshinde** [comment]
@mcastelino yes, we are  doing that so that devices created in the VM's /dev appear in the container /dev.  We filter out initially by mounting /dev as tmpfs so we have a very limited set of devices to begin with  and then use udev events to bind mount all devices that appear after the container has started. We dont have any filtering rules at present for the bind-mounting, these would be any devices that are created after the container has started mostly due to the user interacting on the container command line.

**Comment by amshinde** [comment]
@mcastelino For Kata,  plan is to pass some predictable identifiers for the VFIO devices specifically so that we do not need to pass -v /dev.  I am looking at things such as vendor-id that can be leveraged to identify the VFIO device passed, so that the corresponding device node can be created inside the container. Still looking at this.

File: docs/Passing-Devices-with-Clear-Containers.md [hunk 1]
```diff
@@ -29,6 +29,15 @@ to `vfio-pci` driver and then pass the IOMMU group that the device belongs to on
...
 For detailed steps to bind and unbind devices, see https://github.com/containers/virtcontainers#how-to-pass-a-device-using-vfio-passthrough
 
+You can verify that the device is passed to the Clear Containers container by running
+`lspci` command inside the container.
+
+For the device to be visible inside the container under /dev, you need to pass /dev
+while starting a container as shown below:
+
+```
+$ sudo docker run -it -v /dev:/dev --device=/dev/vfio/16 centos/tools bash
```
Comment (egernst) [inline docs/Passing-Devices-with-Clear-Containers.md]: I think this should be combined with the initial CLI example, and then perhaps explained as you did above.  

There's no usefulness in passing it into the VM, but not the container, after all.

Comment (mcastelino) [reply docs/Passing-Devices-with-Clear-Containers.md]: @amshinde per OCI spec, the devices are container specific resources. So this approach will work fine for docker containers. How do we plan to handle this for kubernetes PODs, where the devices may be spread across containers (i.e. each device belongs to a different container for example)

Comment (amshinde) [reply docs/Passing-Devices-with-Clear-Containers.md]: @mcastelino Containers in a k8 pod should share the same mount namespace eventually.

**Pull request closed without merging**